component-runtime-impl/src/main/java/org/talend/sdk/component/runtime/input/InputImpl.java

```diff
@@ -56,21 +72,78 @@ protected InputImpl() {
         // no-op
     }
 
+    @Override
+    public void start() {
+        super.start();
+        initCheckpointFunctions();
+    }
+
+    @Override
+    public void start(final Consumer checkpointConsumer) {
+        start();
+        if (checkpointEnabled) {
+            checkpointCallback = checkpointConsumer;
+        }
+    }
+
+    protected void initCheckpointFunctions() {
+        if (checkpointEnabled) {
+            checkpoint = findMethods(CheckpointData.class).findFirst().orElse(null);
```

Comment:
those method are optional or we should throw an exception if they are missing? (still reading the code...)